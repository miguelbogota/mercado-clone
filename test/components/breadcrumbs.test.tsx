import Breadcrumbs from '@app-components/breadcrumbs';
import { render, screen } from '@testing-library/react';

describe('Components - Breadcrumbs', () => {
  const items = ['Home', 'Categories', 'Electronics'];

  it('should render correctly', () => {
    const { container } = render(<Breadcrumbs items={items} />);
    expect(container).toMatchSnapshot();
  });

  it('should render all the items', () => {
    render(<Breadcrumbs items={items} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Categories')).toBeInTheDocument();
    expect(screen.getByText('Electronics')).toBeInTheDocument();
  });

  it('should render the correct number of items', () => {
    render(<Breadcrumbs items={items} />);
    expect(screen.getAllByRole('link').length).toBe(items.length);
  });

  it('should render the correct number of separators', () => {
    render(<Breadcrumbs items={items} />);
    expect(screen.getAllByRole('separator').length).toBe(items.length - 1);
  });
});
