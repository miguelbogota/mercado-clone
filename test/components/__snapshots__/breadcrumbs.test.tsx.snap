// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Breadcrumbs should render correctly 1`] = `
<div>
  <div
    class="css-sjgsgk"
  >
    <span
      class="css-1bmx41k"
      role="link"
    >
      <p>
        Home
      </p>
    </span>
    <span
      class="css-1bmx41k"
      role="link"
    >
      <span
        class="material-symbols-outlined"
        role="separator"
      >
        chevron_right
      </span>
      <p>
        Categories
      </p>
    </span>
    <span
      class="css-1bmx41k"
      role="link"
    >
      <span
        class="material-symbols-outlined"
        role="separator"
      >
        chevron_right
      </span>
      <p>
        Electronics
      </p>
    </span>
  </div>
</div>
`;
